app/services/order-service.js

```diff
@@ -7,6 +7,18 @@ export default ApiBaseService.extend({
     return this.GET(`/users/${userId}/orders_count`);
   },
 
+  fetchShipmentorCarryoutCode(detail_type) {
+    return this.GET("/fetch_shipment_or_carryout_code", {
```

Comment:
you can just return the promise
`return this.GET("/fetch_shipment_or_carryout_code", { detail_type: detail_type })`
or
you can return the resolved data.
no need to return both